### ISSUE TITLE
docker-in-docker does not have a non-root writable /dev/null

### ISSUE THREAD

**Issue by technicalpickles** [microsoft/vscode-dev-containers#830]
The other day, I was doing some testing running in the Codespaces  by inheriting from  `codespace-linux` kitchen sink image, and I was having when I got to a console:

```
👋 Welcome to Codespaces! You are on our default image. It includes runtimes and tools for: 
   - Python 
   - Node.js
   - Docker
   - And more! To see the full list, go here: https://aka.ms/ghcs-default-image
🔍 To explore VS Code to its fullest, search using the Command Palette (Cmd/Ctrl + Shift + P or F1).
📝 Edit away, run your app as usual, and we'll automatically make it available for you to access.
/etc/bash.bashrc: line 85: cannot redirect standard input from /dev/null: Permission denied
bash: /dev/null: Permission denied
bash: /dev/null: Permission denied
/usr/local/bin/rbenv: line 22: /dev/null: Permission denied
/usr/local/bin/rbenv: line 31: /dev/null: Permission denied
rbenv: cannot find readlink - are you missing GNU coreutils?
```

I had mentioned this to some other GitHub folks, and was advised not to use that, and instead to use the `debian` image. Well, fast forward a day, and I'm running into the same problem in a new image I based on that. I think I have been able to isolate it though:

- clone this repository
- `cd containers/docker-in-docker/.devcontainer`
- `docker build .`
- `docker run --user vscode  --rm -it --init --privileged <built image> /bin/bash`

You get output like:

```
Not setting dockerd DNS manually.
/usr/local/share/docker-init.sh: line 66: cannot redirect standard input from /dev/null: Permission denied
/etc/bash.bashrc: line 72: cannot redirect standard input from /dev/null: Permission denied
bash: /dev/null: Permission denied
```

And I can confirm it's not writable by the user:

```
vscode ➜ / $ ls -l /dev/null 
crw-rw---- 1 root root 1, 3 Apr 23 19:39 /dev/null
bash: /dev/null: Permission denied
```

**Comment by jkeech** [comment]
cc @joshspicer

**Comment by Chuxel** [comment]
@technicalpickles 

> cd containers/docker-in-docker

I assume you mean `cd containers/docker-in-docker/.devcontainer` ?

If so, I'm not able to reproduce this on my mac. I have no idea why /dev/null would be reacting that way.  Are you doing this in a codespace?  If not can you tell us about your machine setup?

**Comment by technicalpickles** [comment]
@Chuxel that's right about .devcontainer. I just udpated the notes.

I'm running this locally in VS Code stable, on macOS Big Sur (11.2.3), with Docker desktop 3.3.0. I was getting the error originally in VS Code while lauching, but I reproduced it outside of VS Code the devcontainer lifecycle.

One hint I found was https://forums.docker.com/t/selinux-no-longer-able-to-access-dev-in-containers-after-docker-engine-1-11-on-centos-7/9954 but I'm not sure how related it is yet.
